pkg/middleware/requestid/requestid.go

```diff
@@ -15,7 +15,7 @@ import (
 const (
 	requestIDCtxKey   = "request-id-context-key"
 	requestIDTraceKey = "request_id"
-	requestIDHeader   = "x-request-id"
+	RequestIDHeader   = "X-Request-Id"
```

Comment:
These should get canonicalized at some point, but curious what lead to the change?

Reply:
There is no functional change. This is how headers are set:

<img width="882" alt="image" src="https://github.com/openfga/openfga/assets/5374887/dabe43f1-07fb-4c35-90ee-f7cf9e445b48">

Reply:
oh i wonder if assertions should just be against `response.Header.Get(...)`